components/Activitylist.js

```diff
@@ -51,6 +95,16 @@ export default function Activitylist({
               <StyledDeleteButton onClick={() => handleDelete(activity.id)}>
                 <StyledTrashIcon icon={faTrash} />
               </StyledDeleteButton>
+              <StyledFavoriteButton
+                onClick={() => handleAddFavoriteButton(activity.id)}
+              >
+                {activity.likedByUser.some((user) => user.userID === userID) ? (
+                  <StyledFavoriteIconSolid icon={faHeartSolid} />
+                ) : (
+                  <StyledFavoriteIconRegular icon={faHeartRegular} />
+                )}
+                <p>{activity.likedByUser.length}</p>
+              </StyledFavoriteButton>
```

Comment:
coool, i will need a similar function soon, maybe I should get 'very inspired' :P 
(don't worry, ill credit you )